lib/src/robot/sessions_client.dart

```diff
@@ -0,0 +1,98 @@
+import 'dart:async';
+
+import 'package:grpc/grpc_connection_interface.dart';
+import 'package:logger/logger.dart';
+import 'package:viam_sdk/protos/robot/robot.dart';
+
+import '../resource/base.dart';
+
+final _logger = Logger();
+
+/// A Session allows a client to express that it is actively connected
+/// and supports stopping actuating components when it's not.
+class SessionsClient implements ResourceRPCClient {
+  static const sessionMetadataKey = 'viam-sid';
+  static const unallowedMethods = [
```

Comment:
(minor/nit) I think making this a `set` would be more performant given how it's used?

Reply:
Thats a great call out 👀 I will look into this. Thank you!